### ISSUE TITLE
[api-extractor] 7.30.0 causes tryGetEntityForIdentifier() called for an identifier that was not analyzed

### ISSUE THREAD

**Issue by ocavue** [microsoft/rushstack#3620]
## Summary

After updating `api-extractor` from v7.29.5 to v7.30.0, `api-extractor run` will print the following error:

```
ERROR: Internal Error: tryGetEntityForIdentifier() called for an identifier that was not analyzed
```

## Repro steps

I've created a mini repo: https://github.com/issueset/api-extractor-7-30-0-issue

```bash
$ git clone https://github.com/issueset/api-extractor-7-30-0-issue
$ cd api-extractor-7-30-0-issue
$ pnpm install 
$ pnpm build

> api-extractor-issue@1.0.0 build /private/tmp/api-extractor-7-30-0-issue
> tsc -b tsconfig.json && api-extractor run --local


api-extractor 7.30.0  - https://api-extractor.com/

Using configuration from ./api-extractor.json
Analysis will use the bundled TypeScript version 4.7.4

ERROR: Internal Error: tryGetEntityForIdentifier() called for an identifier that was not analyzed
You have encountered a software defect. Please consider reporting the issue to the maintainers of this application.
 ELIFECYCLE  Command failed with exit code 1.
```

Version v7.29.5 is fine. 

```
$ git reset --hard v7.29.5
$ pnpm install
$ pnpm build

> api-extractor-issue@1.0.0 build /private/tmp/api-extractor-7-30-0-issue
> tsc -b tsconfig.json && api-extractor run --local


api-extractor 7.29.5  - https://api-extractor.com/

Using configuration from ./api-extractor.json
Analysis will use the bundled TypeScript version 4.7.4
Warning: The API report file was missing, so a new file was created. Please add this file to Git:
/private/tmp/api-extractor-7-30-0-issue/etc/api-extractor-issue.api.md

API Extractor completed successfully
```

  **Expected result:** No error 

  **Actual result:** Internal Error

## Details

There is a global namespace in the source code, which could cause this bug. 

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/api-extractor` version? | 7.30.0 |
| Operating system? | mac |
| API Extractor scenario? | docs |
| Would you consider contributing a PR? | No |
| TypeScript compiler version? | 4.7.4 |
| Node.js version (`node -v`)? | v16.17.0 |

**Comment by sranka** [comment]
The same error happens in https://github.com/influxdata/influxdb-client-js/pull/565

```
$ api-extractor run
api-extractor 7.30.0  - https://api-extractor.com/

Using configuration from ./api-extractor.json
Analysis will use the bundled TypeScript version 4.7.4
*** The target project appears to use TypeScript 4.8.2 which is newer than the bundled compiler engine; consider upgrading API Extractor.

ERROR: Internal Error: tryGetEntityForIdentifier() called for an identifier that was not analyzed
You have encountered a software defect. Please consider reporting the issue to the maintainers of this application.
error Command failed with exit code 1.
```

**Comment by octogonz** [comment]
@zelliott could this be a regression from https://github.com/microsoft/rushstack/pull/3584 ?

**Comment by zelliott** [comment]
Likely, yes. I'll look into it tomorrow. @ocavue - if you're interested in looking into this before then, see the conversation at https://github.com/microsoft/rushstack/pull/3584#discussion_r954397652 and the associated highlighted code.

**Comment by zelliott** [comment]
We recently released API Extractor v7.31.0. I tried to repro this issue with v7.31.0, and was unable to do so. I think this could be fixed now - @ocavue can you confirm?

**Comment by ocavue** [comment]
@zelliott It's fixed on my side too. Thank you for your help!
